src/runtime_src/core/pcie/driver/linux/xocl/xocl_drm.h

```diff
@@ -47,20 +49,49 @@ struct xocl_cma_bank {
 	struct xocl_cma_memory	cma_mem[1];
 };
 
+struct xocl_mem_group_info {
+        uint32_t l_bank_idx;
+        uint64_t l_start_addr;
+        uint32_t h_bank_idx;
+        uint64_t h_end_addr;
+};
+
+struct xocl_mem_group {
+        int32_t g_count;
+        struct xocl_mem_group_info *m_group[DRM_XOCL_MEM_GROUP_MAX];
+};
+
+struct xocl_mem_map_info {
+        uint32_t cu_id;
+        uint32_t arg_id;
+        uint32_t grp_id;
+};
+
+struct xocl_mem_map {
+        int32_t m_count;
+        struct xocl_mem_map_info *m_map[DRM_XOCL_MEM_GROUP_MAX];
+};
+
+struct xocl_mem_connectivity {
+        struct xocl_mem_map     *mem_map;
+        struct xocl_mem_group   *mem_group;
+};
+
```

Comment:
are these data structure the same as defined in mem_group.h? if so, you need to include the header file here and add comment in the header file to make it super clear that the data structures are shared b/w user space code and driver. you can't have them defined in two places. they could be out of sync later.

Reply:
The mem_group.h file is defined under core/common directory.  Could you please suggest a better place for mem_group.h structure so that both user space and kernel space can share the same file...

Reply:
could you refer to flash_xrt_data.h as an example? that header file is shared b/w user and driver space.

Reply:
I have added the mem_group.h file inside core/common/drv/include/mem_group.h 
Now both userspace and driver are sharing the same data structure.